antarest/core/config.py

```diff
@@ -40,22 +38,38 @@ class InternalMatrixFormat(StrEnum):
     PARQUET = "parquet"
     FEATHER = "feather"
 
-    def load_matrix(self, path: Path) -> npt.NDArray[np.float64]:
-        if self == InternalMatrixFormat.TSV or path.stat().st_size == 0:
-            return np.loadtxt(path, delimiter="\t", dtype=np.float64, ndmin=2)
+    def load_matrix(self, path: Path) -> pd.DataFrame:
+        if path.stat().st_size == 0:
+            return pd.DataFrame()
+        if self == InternalMatrixFormat.TSV:
+            # The legacy format is TSV, so we have to handle both cases
+            # To know if we're opening a legacy matrix or not we have to seek the first bytes of the file
+            # Legacy
+            header = None
+            index_col = None
+            with open(path, "r") as f:
+                if f.read(1) == "\t":
```

Comment:
So this relies on the fact that the top left cell is empty, right ?
I wonder if it's solid enough, in particular for further evolutions that might not be aware of that subtlety. Otherwise we could version matrix format in the database together with its identifier, to decide if it's an old one or a new one.

To be decided